templates/partials/logo.html.twig

```diff
@@ -10,6 +10,27 @@
   {% set logo_img = media['theme://images/grav-logo.svg'] %}
 {% endif %}
 
+{% if theme_var('custom_logo_scrolled') %}
```

Comment:
I'm not really understanding the need for so much complexity here. Couldn't we replace this whole `if` block with something like this:
```suggestion
<a href="{{ home_url }}">
  {{ logo_img.html('', config.site.title, 'logo__img')|raw }}
  
  {% if theme_var('custom_logo_scrolled') %}
    {% set logo_img_scrolled = media['image://webfolio/' ~ (theme_var('custom_logo_scrolled')|first).name] %}

    {{ logo_img_scrolled.html('', config.site.title, 'logo__img__scrolled')|raw }}
  {% endif %}
</a>
```

Then the `.scrolled` class can be used to determine which should be shown/hidden.

(Also, the bit above where I check `image://webfolio/` and `image://images` is just a backwards compatibility thing - no need to do that here.)